azure/datalake/store/lib.py

```diff
@@ -296,8 +143,10 @@ def __init__(self, store_name=default_store, token=None,
 
     @property
     def session(self):
+        bearer_token = self.token_credential.get_token(self.scopes).token
```

Comment:
As you noticed, token_credential could end up with Cli/Pws credential. We should cache it.

Reply:
Added back the caching logic